src/opm/input/eclipse/EclipseState/EclipseState.cpp

```diff
@@ -310,6 +293,45 @@ AquiferConfig load_aquifers(const Deck& deck, const TableManager& tables, NNC& i
             this->aquifer_config.loadFromRestart(aquifers, this->m_tables);
     }
 
+    void EclipseState::assignRunTitle(const Deck& deck)
+    {
+        if (! deck.hasKeyword<ParserKeywords::TITLE>()) {
+            return;
+        }
+
+        const auto& title = deck[ParserKeywords::TITLE::keywordName]
+            .back().getRecord(0).getItem(0);
+
+        this->m_title = fmt::format("{}", fmt::join(title.getData<std::string>(), " "));
+    }
+
+    void EclipseState::reportNumberOfActivePhases() const
+    {
+        const auto nph = this->runspec().phases().size();
+        const auto single_phase = nph == static_cast<std::remove_cv_t<decltype(nph)>>(1);
```

Comment:
This is so very much over-engineered that dummies like me need 10 minutes to get it!
- Please at least use braces around the boolean expression
-  Maybe simply use 1ul instead of the cast? (Actually, I am unable to trick my compiler to warn for `std::size_t nph=1; std::cout<<(nph==1)` using `-Wall -Wextra`, but that might be due to my incompetence.

- Isn't this always 1 (and hence wrong)?
- Why does the type matter at all? It just needs to be convertable to bool

Reply:
>  Please at least use braces around the boolean expression.

Do you mean braces (`{}`) or parentheses (`()`)?  Either way I'm not going to do that.  More punctuation does not make the expression easier to read.  About the only time when I do add those parentheses is when I'm combining logical expressions with an operator such as `&&`.

That said, I've trimmed the expression here down to
```C++
nph == 1
```
since GCC will only warn about signed vs. unsigned comparisons if it is unable to prove that the comparison is safe&mdash;e.g., because both operands are runtime values.  The example program below generates a [diagnostic message (godbolt.org)](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxBJmpAAOqAqETgwe3r56icmOAkEh4SxRMVxxdpgOqUIETMQE6T5%2BXLaY9rkM1bUE%2BWGR0bG2NXUNmc0KQ93BvUX9pQCUtqhexMjsHOYAzMHI3lgA1CYbbk7jxJish9gmGgCCm9u7mAdHyOPoWFSX13e3wQR7LCYwQgc2%2BJgA7FZbnsYXs/nsmIcoTdYXs3iAUMFnlcNtgEUjvqj0SBkgAvTAAfX%2BEQJ0NhxOQWK%2BuL2NI2yKJBHQGLQXn%2BhzcAr2EAizwAIocxQi5s9BUc9mB1gBWNwMRW0u7giW3DgLWicJW8PwcLSkVCcQWWaxopYrJ6bHikAiaXULADWIA2GwAdF6/f6AwA2fScSRGl1mzi8BQgDROl0LOCwGCIFCoFjxOjRciUNAZrMxHaGYBSDRxmi0AjRGMiiMRYK1ACenEdebYggA8gxaM2TbwsICjOI%2B6R8GdKgA3TAxkeYVQVPlrR1/VoR2h4CLEJseLARgjEPAsFu6vgGYAKABqeEwAHcO/FGMeZIIRGJ2FJn/IlGoI7oNvpixQaxrH0DcY0gBZUHidoZwAWg7Mw9lgt5JVMK1LC4cEkIAdTEWgcLnfcmGjVoKnaFwGHcTxGn8SiekKYosiSFIBBGJoEmY9p6L6EoWjaKoJjYvRykqAROjqbiZl48YuiEsYJkkxiuAWBRbVWPR90wJcT31DhDVIY1TXNDhVAADkDWDA0kPYiyMPYpG9DRHOFS0rEsUg9lwQgSAOMwNmaPYPHzehiF8r05l4Z0%2Bzmd1PR9AMEr9YNdLDUgjyVONDN4Yzo1jeNotIJNUzzTMQpzCASoLEASzMZoKyrYgawiOsG2IXtW3TdsCC7HsIwHYth1NMcyLwKcZ1NOcFyrJ8V1001103bcMDWU190PY8FioM9L2vO8H2NR1%2BBfURxA/I6vxUdQR10ZoDCMID0JsBbwJBM1oNSOCEKQlCNjFNC3IsTCcLwgjVCIkj%2BOcCBXDk0hAimBjZg4nJUlh7IWIYRSkZE9pxPqajRj4kaxIUhGeOEwSCfYmSJLJqSJBUtT3ydM5tL1A1wxHYyzIsqy9mAZBkHssxvS4FzgPczz8CIUKHQ8oLSuiMLlMihMFgQc4sBiV6Ut4dLMojHLbDyqKtBi0gPS9X1EoS5LOA2TmjKjfKzcKlMiogJAlgIeI%2BXKyqQtCVg1lUPmWAUCchYnLgAE5vQMKtxl4TBpZIPBuTq2RX1O6RzsUS7f09Ugby3eINpDPTHeyzgOz5X3/lQKg9h5yzrNs4B7MkRznIgBWCzCjYIpd10Lbi62bf9CvUv1gzDedmM41Nk8Pa9uu/YoCr00V4gg7YTgW75gWhdKUXk9Tg8M8/bP31z2R85/a69BLpgy%2B4HSOdnrma7Xhum4P6yj7C1Fi5Le/cHRDyXubDWTAtaUHZhwaeIAMqfydhwXKi81ajythPAMFcHYoOrmg4e5tdJmCrpGIhkCFhTkaqkEAkggA%3D) at `-Wall -Wextra` for "signed vs. unsigned" comparisons.

---

```C++
#include <iostream>
#include <cstddef>
int main()
{
    int a;
    std::cin >> a;
    std::size_t b;
    std::cin >> b;
    std::cout << (b == a) << '\n';
}
```